MetaMask/metamask-extension
# feat: UX: Multichain: Signal to dapp manual network changes

## **Description**

This PR signals to a dapp when the user has manually changed network during a dapp context.  The result is the dapp switching to the network that the user manually changed to.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23809?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2087

## **Manual testing steps**

1. Go to Uniswap
2. Connect your wallet
3.  Switch to Polygon network via the Uniswap dapp
4. Manually open the MetaMask popup, see Polygon selected
5. Manually change the network to Ethereum Mainnet using the network picker
6. See the network change in the wallet to Ethereum Mainnet
7. Close the popup, see that Uniswap has changed to Ethereum Mainnet as well

## **Screenshots/Recordings**



### **Before**

N/A

### **After**


https://github.com/MetaMask/metamask-extension/assets/46655/24b1d4dd-431d-46c8-b7c1-fc5c0f788273



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
